src/app/report/edit-display/edit-display.component.ts

```diff
@@ -209,20 +195,28 @@ export class EditDisplayComponent {
       checkpointId = this.report.uid.split('#')[1];
     }
 
-    const body = { stub: stubStrategy, ...this.getReportValues(checkpointId) };
+    const body = this.getReportValues(checkpointId);
 
     this.httpService.updateReport(storageId, body, this.currentView.storageName).subscribe({
       next: (response: any) => {
         response.report.xml = response.xml;
         this.report = response.report;
         this.saveReportEvent.next(this.report);
-        this.editor.setNewReport(this.report.xml);
+        this.editor.setNewReport(UpdateReportUtil.isUpdateCheckpoint(body) ? body.checkpointMessage : this.report.xml);
```

Comment:
Seeing as setNewReport only needs a string, couldn't you just use:
```suggestion
        this.editor.setNewReport(body.checkpointMessage ?? this.report.xml);
```

Reply:
No, because body is from type UpdateReport | UpdateCheckpoint, so Typescript doesn't know if property checkpoinMessage exists without the UpdateReportUtil function